Add to vendors_default the possibility to add values for IstioCNI (#845)
istio-ecosystem/sail-operator: tests/e2e/README.md

To do this, we have a set of test cases under a test group called `smoke`. This test group will run a set of tests that are designed to test the basic functionality of the operator. The tests in this group are designed to be run against a cluster that is already configured and running. The tests will not modify the cluster configuration or the operator configuration. The tests will only check if the operator is working as expected.

Pre-requisites:
* The operator is already installed and running in the cluster.

To run the test group, you can use the following command:

* Run the following command to run the smoke tests:
For running on kind:
```
$ SKIP_BUILD=true SKIP_DEPLOY=true GINKGO_FLAGS="-v --label-filter=smoke" make test.e2e.kind
```

For running on OCP:
```
$ SKIP_BUILD=true SKIP_DEPLOY=true GINKGO_FLAGS="-v --label-filter=smoke" make test.e2e.ocp
```

#### Running with specific configuration for the Istio and IstioCNI resource

> It can be cases where you want to run the test with a specific configuration for the Istio and IstioCNI resource to match some cluster specific needs. For this, you can modify the `pkg/istiovalues/vendor_default.yaml`file to default `spec.values` for the Istio and IstioCNI resources. 

Reviewer: ```suggestion
There might be situations where you want to run tests with a specific configuration for the Istio and IstioCNI resource to match some cluster specific needs. For this, you can modify the `pkg/istiovalues/vendor_default.yaml`file to default `spec.values` for the Istio and IstioCNI resources. 
```

> This file will be used by the operator to set specific `spec.values` for the Istio and IstioCNI resources. Example:

Reviewer: we're already explaining this in the vendor_defaults.yaml file, should we just link to that? I'm worried we're duplicating information with the risk of one getting outdated

      - https://istio-release.storage.googleapis.com/charts/ztunnel-1.26.0.tgz
  - name: v1.25-latest
    ref: v1.25.3
  - name: v1.25.3
    version: 1.25.3
    repo: https://github.com/istio/istio
    commit: 1.25.3
    charts:
      - https://istio-release.storage.googleapis.com/charts/base-1.25.3.tgz
      - https://istio-release.storage.googleapis.com/charts/istiod-1.25.3.tgz
      - https://istio-release.storage.googleapis.com/charts/gateway-1.25.3.tgz
      - https://istio-release.storage.googleapis.com/charts/cni-1.25.3.tgz
      - https://istio-release.storage.googleapis.com/charts/ztunnel-1.25.3.tgz
```
*Important*: avoid adding in the custom file versions that are not available in the `pkg/istioversion/versions.yaml` file. The test framework will not be able to run the tests because the operator does not contains the charts for those versions.
* To run the test framework against a specific Istio version, you can use the following command:
```
$ VERSIONS_YAML_FILE=path/to/your/versions.yaml SKIP_BUILD=true SKIP_DEPLOY=true GINKGO_FLAGS="-v --label-filter=smoke" make test.e2e.kind
```

### Understanding the test output

> By default, if you run the test using the make target, it will be generated a `report.xml` file the root directoyry of the project. This file contains the test results in JUnit format. For example the file will contain the following:

Reviewer 2: Could you rephrase the sentence?

Everything else looks good.
